File: lingting-spring-mybatis/src/main/java/live/lingting/spring/mybatis/ExtendServiceImplBeanPost.java
package live.lingting.spring.mybatis;

import live.lingting.framework.mybatis.extend.ExtendMapper;
import live.lingting.framework.mybatis.extend.ExtendServiceImpl;
import live.lingting.spring.post.SpringBeanPostProcessor;
import live.lingting.spring.util.SpringUtils;
import lombok.RequiredArgsConstructor;

/**
 * @author lingting 2024-03-12 17:41
 */
@RequiredArgsConstructor
@SuppressWarnings("java:S3740")
public class ExtendServiceImplBeanPost implements SpringBeanPostProcessor {

	@Override
	public boolean isProcess(Object bean, String beanName, boolean isBefore) {
		return bean instanceof ExtendServiceImpl;
	}

	@Override
	public Object postProcessAfter(Object bean, String beanName) {
		if (bean instanceof ExtendServiceImpl impl) {
			Class mapperClass = impl.getMapperClass();
			Object mapper = SpringUtils.getBean(mapperClass);
			impl.setMapper((ExtendMapper) mapper);
		}
		return bean;
	}

}

File: lingting-spring-mybatis/src/main/kotlin/live/lingting/spring/mybatis/ExtendServiceImplBeanPost.kt
package live.lingting.spring.mybatis

import live.lingting.framework.mybatis.extend.ExtendMapper
import live.lingting.framework.mybatis.extend.ExtendServiceImpl
import live.lingting.spring.post.SpringBeanPostProcessor
import live.lingting.spring.util.SpringUtils

/**
 * @author lingting 2024-03-12 17:41
 */
@Suppress("UNCHECKED_CAST")
class ExtendServiceImplBeanPost : SpringBeanPostProcessor {

    override fun isProcess(bean: Any, beanName: String, isBefore: Boolean): Boolean {
        return bean is ExtendServiceImpl<*, *>
    }

    override fun postProcessAfter(bean: Any, beanName: String): Any {
        if (bean is ExtendServiceImpl<*, *>) {
            val impl = bean as ExtendServiceImpl<ExtendMapper<Any>, Any>
            val mapperClass = impl.mapperClass
            val mapper = SpringUtils.getBean(mapperClass)
            impl.mapper = mapper
        }
        return bean
    }
}
